src/index.js

```diff
@@ -549,9 +550,16 @@ function createConfig(options, entry, format, writeMeta) {
 									...(options.generateTypes !== false && {
 										declarationDir: getDeclarationDir({ options, pkg }),
 									}),
-									jsx: 'preserve',
-									jsxFactory: options.jsx,
-									jsxFragmentFactory: options.jsxFragment,
+									...(options.jsxImportSource
+										? {
+												jsx: 'react-jsx',
+												jsxImportSource: options.jsxImportSource,
+										  }
+										: {
+												jsx: 'preserve',
+												jsxFactory: options.jsx,
+												jsxFragmentFactory: options.jsxFragment,
+										  }),
```

Comment:
Theoretically we would be fine to leave `jsx` untouched & remove that first piece of the ternary, however, that relies on the assumption that the user does have a `tsconfig.json` and that it's correctly set.

Conditionally adding `jsxFactory` (and `jsxFragmentFactory`) is enough to quell the errors we were getting in the linked issue, but when using `--jsxImportSource` with TS, the `tsconfig.json` will need to have different values. While users in all likelihood will have a `tsconfig.json` that's correctly configured in that case, if they do not, their typings will contain `any` where JSX references should be.

Setting these configurations ourselves, at least as defaults, allows us to make fewer assumptions about users' configs that are external to Microbundle (and hopefully will raise fewer issues).